src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -299,6 +320,64 @@ comps_init(void)
 	return;
 }
 
+void 
+chkpt_comp_init(struct crt_comp *comp, struct crt_chkpt *chkpt, char *name)
+{
+#ifdef ENABLE_CHKPT
+	/* create the component */
+	void *elf_hdr;
+	int   keylen;
+	compid_t id = crt_ncomp() + 1;
+	const char *root = "binaries/";
+	int   len  = strlen(root);
+	char  path[INITARGS_MAX_PATHNAME];
+
+	assert(id < MAX_NUM_COMPS && id > 0 && name);
+	memset(path, 0, INITARGS_MAX_PATHNAME);
+	strncat(path, root, len);
```

Comment:
before this: `assert(len < INITARGS_MAX_PATHNAME)`? I believe this is productive. Could be wrong and it mioght be redundant with some check somewhere.